### DISCUSSION TITLE
BepInEx 6.0.0-pre.2

### DISCUSSION THREAD

**Discussion by ManlyMarco** [BepInEx/BepInEx#969] [Show and tell]
> **NOTE:** This is a **pre-release** of a major version of BepInEx. BepInEx 5 plugins will not work with this release yet.

### About upgrading

This release is mainly meant for those who use current Bleeding Edge builds to mod Il2Cpp games. If you want to use this release, follow this upgrade guide:

* **If you use BepInEx 5:** Stay on BepInEx 5! The current pre-release does not support loading BepInEx 5 plugins, and there are many issues.
* **If you use BepInEx Bleeding Edge builds (builds 697 down to 577):** It is recommended that you upgrade to this pre-release and stay on it to ensure all your plugins keep working. Builds after 6.0.0-be.697 will bring many breaking API changes in preparation for the first stable v6 release.
* **If you want to mod new Unity Mono games:** Consider using BepInEx 5! The API is still stable, and BepInEx 6 will eventually include support for loading BepInEx 5 plugins!
* **If you want to mod new Il2Cpp games:** Consider using this release until the stable v6 release, or until [bleeding edge builds](https://builds.bepinex.dev/projects/bepinex_be) enter the RC (release candidate) stage.

## Changelog

The changelog is concerning BepInEx 6.0.0-pre.1

### Merged PRs
* Restructure project, add Doorstop 4 support, use CoreCLR for Il2Cpp by @ghorsington in https://github.com/BepInEx/BepInEx/pull/391
* Add chainloader events by @js6pak in https://github.com/BepInEx/BepInEx/pull/475
* Warn about outdated wine by @js6pak in https://github.com/BepInEx/BepInEx/pull/477
* Add a config option to disable xref and disable it by default on x32 by @js6pak in https://github.com/BepInEx/BepInEx/pull/502
* Bump Il2CppInterop by @js6pak in https://github.com/BepInEx/BepInEx/pull/503
* Switch to the `development` branch of Cpp2IL by @ds5678 in https://github.com/BepInEx/BepInEx/pull/533
* Bump Cpp2IL by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/560
* Bump Il2CppInterop to v1.4.4 by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/561
* Bump Il2CppInterop to v1.4.5 by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/564
* Bump Il2CppInterop by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/647
* Don't try to parse the wine version by @js6pak in https://github.com/BepInEx/BepInEx/pull/555
* Fix argument count when launched with Steam on Linux by @zeroepoch in https://github.com/BepInEx/BepInEx/pull/563
* Bump Il2CppInterop & Dobby by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/669
* Fix netfx & netcore build targets, and add target framework variants by @bbepis in https://github.com/BepInEx/BepInEx/pull/715
* Reimplement UnhollowedAssembliesPath config option as IL2CPPInteropAssembliesPath by @Kokaiinum in https://github.com/BepInEx/BepInEx/pull/734
* Fix manager GameObject not having a name by @ManlyMarco in https://github.com/BepInEx/BepInEx/pull/735
* run_bepinex*.sh: self-relaunch improvements by @astiob in https://github.com/BepInEx/BepInEx/pull/692
* Bump Cpp2IL & Il2CppInterop by @Kasuromi in https://github.com/BepInEx/BepInEx/pull/869
* Upgrade doorstop to version 4.2.0 (v6) by @aldelaro5 in https://github.com/BepInEx/BepInEx/pull/865
* Upgrade AssetRipper.Primitives to avoid implying Json 8.0.1 by @decaprime in https://github.com/BepInEx/BepInEx/pull/873
* Upgrade Doorstop to 4.3.0 (v6) by @aldelaro5 in https://github.com/BepInEx/BepInEx/pull/901
* Upgrade Harmonyx, Iced and AssetRipper.Primitives packages by @aldelaro5 in https://github.com/BepInEx/BepInEx/pull/953

### New Contributors
* @ds5678 made their first contribution in https://github.com/BepInEx/BepInEx/pull/533
* @Kokaiinum made their first contribution in https://github.com/BepInEx/BepInEx/pull/734
* @decaprime made their first contribution in https://github.com/BepInEx/BepInEx/pull/873

**Full Changelog**: https://github.com/BepInEx/BepInEx/compare/v6.0.0-pre.1...v6.0.0-pre.2

<hr /><em>This discussion was created from the release <a href='https://github.com/BepInEx/BepInEx/releases/tag/v6.0.0-pre.2'>BepInEx 6.0.0-pre.2</a>.</em>

**Comment by MrOlsen857** [comment]
good

**Reply by Aanonymz** [reply]
wich version should i download for it to support sun haven or does it even support it ? or just only bepin 5 as of now?

**Comment by Cheetachaser** [comment]
As of 5.4... Newer games return this:
Info
--- > System. FormatException: Unsupported metadata version found! We support 23-29, got 31
at LibCpp2IL. Metadata. Il2CppMetadata. ReadFrom(Bytel] bytes, UnityVersion unityVersion) in /home/runner/work/Cpp2IL/Cp
p2IL/LibCpp2IL/Metadata/ Il2CppMetadata.cs: line_71
at LibCpp2IL.LibCpp2IlMain. Initialize(Byte[] binaryBytes, Bytel] metadataBytes, UnityVersion unityVersion) in /home/r
unner/work/Cpp2IL/Cpp2IL/LibCpp2IL/LibCpp2IlMain.cs: line 149
at LibCpp2IL.LibCpp2IlMain. LoadFromFile(String pePath, String metadataPath, UnityVersion unityVersion) in /home/runne
r/work/Cpp2IL/Cpp2IL/LibCpp2IL/LibCpp2IlMain.cs: line 202
at Cpp2IL. Core. Cpp2IlApi. InitializeLibCpp2I|(String assemblyPath, String metadataPath, UnityVersion unityVersion, Boo
ean al lowUserToInputAddresses) in /_/Cpp2IL. Core/Cpp2IlApi.cs: line 62
--- End of inner exception stack trace -
at Cpp2IL. Core. Cpp2IlApi. InitializeLibCpp2I|(String assemblyPath, String metadataPath, UnityVersion unityVersion, Boo
ean al lowUserToInputAddresses) in /_/Cpp2IL. Core/Cpp2IlApi.cs: line 68
at BepInEx. Unity. IL2CPP. Il2CppInteropManager. RunCpp2Il() in /home/runner/work/BepInEx/BepInEx/Runtimes/Unity/BepInEx.
Unity. IL2CPP/Il2CppInteropManager.cs: line 301
at BepInEx. Unity. IL2CPP. Il2CppInteropManager. GenerateInteropAssemblies() in /home/runner/work/BepInEx/BepInEx/Runtime
s/Unity/BepInEx. Unity. IL2CPP/I12CppInteropManager.cs: line 236
: Preloader] 0 patcher plugins loaded
: Preloader] 0 assemblies discovered

The latest RaiPatcher does not work either, though that program does not tell you why, I assume it is a Unity update.

**Reply by simsler-gkbears** [reply]
Same problem here.
Can you help us please?

Kind regards
Simsler

**Comment by pakclia** [comment]
il2cpp the game does not start with BeplnEx, an injection occurs, then a crash, what should I do?

**Comment by H644b** [comment]
Does this bring il2cpp support for mac?

**Reply by toebeann** [reply]
Depends on the game. This is an old release, it will work with older IL2CPP games on mac. For newer games, you probably need a bleeding edge build. Check the readme for links.

**Reply by H644b** [reply]
What about a game like BloonsTD6? Its from Dec 2018.

**Reply by toebeann** [reply]
Without owning the game myself or having personal knowledge of what the community of that game uses to mod it, I couldn't tell you.

Try it. If it doesn't work, try the latest bleeding edge build. If that also doesn't work, then no dice.
